### PR TITLE
chainparams: Update with data from assumed valid chain

### PR DESCRIPTION
**Pull request by maflcko**
Can be reviewed by using the `getblock` and `getchaintxstats` rpcs of a synced node. Reviewers get extra points when their full node has checkpoints and assumevalid disabled.

### CODE REVIEW

**Comment by gmaxwell** [comment]
ACK mainnet work and assumevalid. Didn't check txdata or testnet.

**Comment by laanwj** [comment]
utACK fa6094f1529ccb8068db06c63a0fecebb3f82583
### mainnet
```javascript
user@medea:~ % cli getchaintxstats null "0000000000000000002e63058c023a9a1de233554f28c7b21380b6c9003f36a8"
{
  "time": 1532884444,  // matches chainTxData
  "txcount": 331282217, // matches chainTxData
  "window_final_block_hash": "0000000000000000002e63058c023a9a1de233554f28c7b21380b6c9003f36a8",
  "window_block_count": 4320,
  "window_tx_count": 5864112,
  "window_interval": 2477828,
  "txrate": 2.366634003651585 // matches chainTxData
}
user@medea:~ % cli getblock 0000000000000000002e63058c023a9a1de233554f28c7b21380b6c9003f36a8
{
…
  // matches consensus.nMinimumChainWork
  "chainwork": "0000000000000000000000000000000000000000028822fef1c230963535a90d", 
…
}
```
### testnet
Will fill this in after testnet finishes syncing.

File: src/chainparams.cpp [hunk 1]
```diff
@@ -217,10 +216,10 @@ class CTestNetParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_SEGWIT].nTimeout = 1493596800; // May 1st 2017
 
         // The best chain should have at least this much work.
-        consensus.nMinimumChainWork = uint256S("0x00000000000000000000000000000000000000000000002830dab7f76dbb7d63");
+        consensus.nMinimumChainWork = uint256S("0x00000000000000000000000000000000000000000000007dbe94253893cbd463");
```
Comment (achow101) [inline src/chainparams.cpp]: The chainwork I get at block 1354312 is `00000000000000000000000000000000000000000000006b4a51a422c1537821`. However `getblockchaininfo` reports a higher chainwork (currently at block 1356214) so I guess this is fine.

**Review by achow101 — COMMENTED** [review]
ACK fa6094f1529ccb8068db06c63a0fecebb3f82583

I have verified that the mainnet parameters match the expected values. 

The testnet ones mostly match, except for the chainwork. However I think that is find as `getblockchaininfo` has a higher value for chainwork anyways. I assume that the chainwork value was taken from `getblockchaininfo` at the time of the commit rather than the chainwork of the assumevalid block.

**Comment by MarcoFalke** [comment]
Fixed the block hash in the comment pointed out by @achow101 and pushed as a new "doc" commit to not touch the previous commit/review.

**Comment by laanwj** [comment]
Checked testnet, found only the same mismatch for nMinimumChainWork that @achow101 did, but yes I think that's ok.

**Comment by Sjors** [comment]
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

Post merge ACK fa8f2d826c7ed90c164cfc039d97ce2fa95d1914 for main chain, no bonus points:

bitcoin-cli getchaintxstats null "0000000000000000002e63058c023a9a1de233554f28c7b21380b6c9003f36a8"
{
  "time": 1532884444,
  "txcount": 331282217,
  "window_final_block_hash": "0000000000000000002e63058c023a9a1de233554f28c7b21380b6c9003f36a8",
  "window_block_count": 4320,
  "window_tx_count": 5864112,
  "window_interval": 2477828,
  "txrate": 2.366634003651585
}

bitcoin-cli getblock 0000000000000000002e63058c023a9a1de233554f28c7b21380b6c9003f36a8
...
  "chainwork": "0000000000000000000000000000000000000000028822fef1c230963535a90d",
...
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEE7ZvfetalXiMuhFJCV/+b28wwEAkFAltjEKUACgkQV/+b28ww
EAkXJBAAhUSEBUx8AZfOssrKmQJEAemAsfEbWkQEgg0Uf9h9MtD9rPjXMuyGA/UG
AJKGDo6Dkn9/7db8YojyB+2o47wHOXyhRKqlZDZp3vm5yYWiluBVYw3oN1erN3sh
3M2VxV/VQPahEb93n1QgAsielwHa35sL5UblJnMV2osDk3C/7DHuh2igJHblb2bZ
Wqn//cPyQc4sAjbIxwgAd58M6hvDgMlmkP2d0fwsvuwlQ9H4G22YcN3ev1amAChG
1Fvtw2bbyS9VJewf866gPXFnTqlCWTbkFmKn8Mfr5hFo8jALc7Y0WMW3NgQHs7i/
rDowOJFZbKSQzsfSjgGlUojO68GUQc5P3ZMvT0VcUIi2Y7P8U+HxLfoKef2dXPnH
VXY5n6oysxBzGoKaHWyLvLzftAY43fjzQeXbZ4bsp6cvODKRZWpd0sM2we9d5tru
pSr/PJajP10WOn4C5qPLUCWiTk1A5uTAAgQUkp+vMuiJwgT01DIrFnVbqRKROH5p
Pp3fzsDaVd2tCtsaO7H4wdSgY4PC6D+LnCI02//CEY8TafRiaRm1Kb4wR6lX+stO
/PpbpxJRICjO0Pa64U8RtZ/1Q3lejGzu4FIuYMVUz3485EgFE41v4B0tls+UJTJe
OVh7quUiP8JHeSgjTnX0dLWu9HtiOTxB76QzINHONaStPUbILzE=
=wzx0
-----END PGP SIGNATURE-----
```

**Pull request merged**